pkg/cmd/ssh/options.go

```diff
@@ -463,11 +464,17 @@ func (o *SSHOptions) Run(f util.Factory) error {
 		return err
 	}
 
-	// if a node was given, check if the node exists
-	// and if not, exit early and do not create a bastion
-	node, err := getShootNode(ctx, o, shootClient, currentTarget)
-	if err != nil {
-		return err
+	var nodeHostname string
+	if o.NodeName != "" {
+		nodeHostname, err = getNodeHostname(ctx, o, shootClient, o.NodeName)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return fmt.Errorf("failed to determine hostname for node: %w", err)
+			}
+
+			fmt.Fprintf(o.IOStreams.Out, "Node with name %q not found. Wrong name provided or this node did not yet join the cluster, continuing anyways: %v\n", o.NodeName, err)
+			nodeHostname = o.NodeName
+		}
```

Comment:
```suggestion
	var nodeHostname string
	if o.NodeName != "" {
          node, err := getShootNode(ctx, o, shootClient, currentTarget)
          if err == nil {
            nodeHostname, err = getNodeHostname(node)
            if err != nil {
              return err
            }
          } else if apierrors.IsNotFound(err) {
            fmt.Fprintf(o.IOStreams.Out, "Node with name %q not found. Wrong name provided or this node did not yet join the cluster, continuing anyways: %v\n", o.NodeName, err)
            nodeHostname = o.NodeName
          } else {
            return fmt.Errorf("failed to determine hostname for node: %w", err)
          }
```

I think it would be more clear to not mix `getNodeHostname` and `getShootNode`

Reply:
I not sure if this proposal is correct but it should give you the idea.

Reply:
✅